File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        int n =0;
        
        ListNode t=head;
        while(t!=null){
            n++;
            t=t.next;
        }
        if(n==0|| n==1){
          return null;
        }
        int c=(n/2)+1;
        t=head;
        ListNode prev=head;
        while(c>1){
           prev=t;
           t=t.next;
           c--;
        }
        prev.next=t.next;
        return head;
    }
}

File: 2095-delete-the-middle-node-of-a-linked-list/2095-delete-the-middle-node-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        int c=0;
        ListNode* t = head;
        while( t != nullptr){
            c++;
            t=t->next;
        }
        if(c==1) return head ->next;
        ListNode* s=head,*f=head ,*p;
        if(c%2==0){
            while(f!=nullptr && f->next->next != nullptr){
                s=s->next;f=f->next->next;
            }
            s->next = s->next ->next;
        }
        else{
            while(f->next != nullptr){
                p=s;
                s=s->next;
                f=f->next->next;
            }
            p->next=s->next;
        }
        return head;
    }
};